--- /home/james/tinyos-main/tos/lib/net/4bitle/LinkEstimatorP.nc	2014-02-21 23:04:38.137322195 +0000
+++ LinkEstimatorP.nc	2014-04-15 14:20:10.049915712 +0100
@@ -226,6 +226,11 @@
 	dbg("LI", "Pinned entry, so continuing\n");
 	continue;
       }
+      //if cluster head ignore
+      if ((NeighborTable[i].ll_addr) >= CLUSTER_HEAD_MIN_ID) {
+	dbg("LI", "Cluster Head entry, so continuing\n");
+	continue;
+      }	
       thisETX = NeighborTable[i].etx;
       if (thisETX >= worstETX) {
 	worstNeighborIdx = i;
@@ -444,6 +449,11 @@
   // return bi-directional link quality to the neighbor
   command uint16_t LinkEstimator.getLinkQuality(am_addr_t neighbor) {
     uint8_t idx;
+    //return max etx value if its a leaf path
+    if (!CLUSTER_HEAD && ((neighbor) <  CLUSTER_HEAD_MIN_ID)){
+      dbg("LI", "updateNeighborTableEst: Set etx to max where leaf path\n");
+      return VERY_LARGE_ETX_VALUE;
+    }
     idx = findIdx(neighbor);
     if (idx == INVALID_RVAL) {
       return VERY_LARGE_ETX_VALUE;
